mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraDeletedMessageDAO.java

```diff
@@ -123,12 +127,47 @@ public Mono<Void> addDeleted(CassandraId cassandraId, MessageUid uid) {
                 .setLong(UID, uid.asLong()));
     }
 
+    public Mono<Void> addDeleted(CassandraId cassandraId, Collection<MessageUid> uids) {
+        if (uids.size() == 1) {
+            return cassandraAsyncExecutor.executeVoid(
+                addStatement.bind()
+                    .setUUID(MAILBOX_ID, cassandraId.asUuid())
+                    .setLong(UID, uids.iterator().next().asLong()));
+        } else {
+            return Flux.fromIterable(uids)
+                .window(BATCH_STATEMENT_WINDOW)
+                .flatMap(partialUidsFlux -> partialUidsFlux.reduce(new BatchStatement(BatchStatement.Type.UNLOGGED), (batch, uid) ->
+                    batch.add(addStatement.bind()
+                        .setUUID(MAILBOX_ID, cassandraId.asUuid())
+                        .setLong(UID, uid.asLong()))))
```

Comment:
We IMO do not need a Flux to build the batch statements.

```
Stream<BatchStatement> batches = Lists.partition(uids, 1024)
    .stream()
    .map(uidBatch -> {
        BatchStatement batch = new BatchStatement(BatchStatement.Type.UNLOGGED);
        uidBatch.forEach(uid -> batch.add(addStatement.bind()
                        .setUUID(MAILBOX_ID, cassandraId.asUuid())
                        .setLong(UID, uid.asLong()))));
        return batch;
    });
return Flux.fromStream(batches)
    .concatMap(cassandraAsyncExecutor::executeVoid)
    .then();
```

Rationals: being asynchronous is not needed when building the batches... `window` operator likely is expensive.